proposals/solid-notifications.md

```diff
@@ -0,0 +1,270 @@
+1. **Avoiding missing updates** - for WebSockets and protocols that rely on an active, live connection to a notification server, the notification protocol needs to make sure that clients do not miss notifications in the event of a dropped connection.
+1. **Protocol negotiation** - a given Solid Notification server may support certain technologies (e.g. WebSockets and LDN) but not others (e.g. EventSource and WebSub). Likewise, a client may not support the same set of protocols that are implemented by a server. As such, there needs to be a mechanism for clients and servers to agree on a mutually supported protocol. In addition to simply determining the set of protocols that work for client and server, there may be particular features (e.g. notification aggregation, notification filtering) that are required for the client. This could be used to further filter the protocol selection.
+1. **Security** - a client should not be able to subscribe to resources to which it does not have read access.
+
+## Terminology
+
+This document uses terms from the Solid Protocol specification, including "data pod". This document also uses terms from the OAuth2 specification, including "resource server", "authorization server" and "client", as well as terms from the WebSub specification, including "topic".
+
+In addition, the following terms are defined:
+
+**Notification Gateway API** -- an HTTP endpoint at which a client can negotiate an acceptable notification subscription location
+
+**Notification Subscription API** -- an HTTP endpoint at which a client can initiate a subscription to notification events for a particular set of resources
+
+**Solid Server Metadata Resource** - an RDF document that includes metadata about the Solid server
+
+## High-level Flow
+
+The following diagram shows the high level interactions involved in this flow. How a client retrieves an access token for step 5 is outside the scope of this document.
+
+![High level flow](solid-notifications-flow.png)
+
+## Discovery
+
+For any resource in a data pod, a client can discover the associated notification gateway API by fetching the Solid Server Metadata Resource. This resource can be retrieved by appending /.well-known/solid to the base URL of the data pod.
+
+```http
+GET /.well-known/solid
+```
+
+The server `MUST` be capable of serializing this resource as Turtle or JSON-LD.
+
+A sample representation of this resource might include the following:
+
+```http
+Content-Type: application/ld+json
+
+{
+    "@context": ["https://www.w3.org/ns/solid/notification/v1"],
+    "notification_endpoint": "https://gateway.example",
+    ...
+}
+```
+
+**Note**: the notification panel should define a particular JSON-LD @context for use with JSON-LD serializations.
+
+**Note**: the use of a well-known resource is based on suggestions to avoid too many link headers, but that is not a settled issue.
```

Comment:
Just a general comment that I like this use of .well-known as a discovery mechanism for server support. It has broad adoption in other protocols, fits nicely here, and would useful for other things, like whether a server supports / enforces data validation, etc.